network/transport/grpc/authenticator.go

```diff
@@ -55,6 +55,7 @@ func (t tlsAuthenticator) Authenticate(nodeDID did.DID, grpcPeer grpcPeer.Peer,
 				WithError(err).
 				Warn("Connection manually authenticated with authentication error")
 			peer.NodeDID = nodeDID
+			peer.Authenticated = false
```

Comment:
Default would be false already. Is there a way to go from true to false in this flow?

Reply:
Yes, see last commit. However, I just added this for explicitness/readability.